### ISSUE TITLE
ImageDataGenerator complains about lack of stratification but I'm doing regression

### ISSUE THREAD

**Issue by beeb** [tensorflow/tensorflow#38625]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): `yes` but it's basic
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): `macOS 10.15.4`
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): `binary`
- TensorFlow version (use command below): `v2.1.0-rc2-17-ge5bf8de410 2.1.0`
- Python version: `3.7.7`
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Describe the current behavior**

When using `ImageDataGenerator.flow()` with a `validation_split` argument, a `ValueError` exception is raised that states:

> Training and validation subsets have different number of classes after the split. If your numpy arrays are sorted by the label, you might want to shuffle them.

However, my labels data is continuous (floats) and I'm doing regression.

**Describe the expected behavior**

This code should not throw. At the very least, it should be a warning and I should be able to silence it.

**Standalone code to reproduce the issue**

```python
y = 500 * numpy.random.rand(200)  # labels are float values
datagen = ImageDataGenerator(validation_split=0.2)
gen_train = datagen.flow(
    x, y, shuffle=True, subset='training'
)
```

**Comment by beeb** [comment]
Looks like a problem in `keras-preprocessing` which is already reported here https://github.com/keras-team/keras-preprocessing/issues/214
